azure/ConsiderationsForServiceDesign.md

```diff
@@ -77,6 +77,65 @@ Understanding how your service is used and defining its model and interaction pa
 
 :white_check_mark: **DO** create an [OpenAPI Definition](https://github.com/OAI/OpenAPI-Specification/blob/main/versions/2.0.md) (with [autorest extensions](https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md)) describing the service. The OpenAPI definition is a key element of the Azure SDK plan and is essential for documentation, usability and discoverability of services.
 
+## Use Good Names
+
+Good names for resources, properties, operations, and parameters are essential to a good developer experience.
+
+Resources are typically described by nouns. Resource and property names must be descriptive and easy to understand for end-users. Ease of understanding comes from familiarity and recognition; you should favor consistency with other Azure services, names in the product user interface, and industry standards.
+
+Names should aid developers in discovering functionality without having to constantly refer to documentation.
+Use common patterns and standard conventions to aid developers in correctly guessing common property names and meanings.
+Use verbose naming patterns and avoid abbreviations other than
+well-known acronyms in your service domain.
+
+### Recommended Naming Conventions
+
+The following are recommended naming conventions for Azure services:
+
+:heavy_check_mark: **DO** use singular nouns for resource (schema) names.
+
+:heavy_check_mark: **DO** use plural nouns for collections (for listing type or collection properties).
```

Comment:
I would suggest that we use a page suffix or similar rather than pluralizing the name. If nothing else, it will make it possible to provide a reusable pattern in Cadl that doesn't require the user to provide both a singular and plural name for a given resource type.

Reply:
A resource will only have a singular name. Only a collection of resources will have a plural name.

Reply:
Are you referring to path segments or model names here? Because I'm referring to the model name for the response model for a list operation? If we require model names to use plural, the effect is that we'll require Cadl authors to specify the plural name explicitly for a given resource. I would prefer that we don't and instead use `PageOfThing` rather than `Things`.

Reply:
I can see that this is unclear.  I think this mainly refers to property names of arrays or maps. But I don't love `PageOfThing` as a schema name. A good item for discussion.

Reply:
I'll remove guidance on schema names. We can put that in the Azure API Style Guide.